python/looker_sdk/rtl/serialize.py

```diff
@@ -96,21 +91,41 @@ def serialize(api_model: TModelOrSequence) -> bytes:
     return json.dumps(data).encode("utf-8")  # type: ignore
 
 
-def structure_hook(context, converter, data, type_):
-    """cattr structure hook used in generated models.
+def _tr_data_keys(data):
+    """Map top level json keys to model property names.
+
+    Currently this translates reserved python keywords like "from" => "from_"
+    """
+    for reserved in keyword.kwlist:
+        if reserved in data and isinstance(data, dict):
+            data[f"{reserved}_"] = data.pop(reserved)
+    return data
+
+
+def translate_keys_structure_hook(converter, data, model_type):
+    """Applied only to models.Model
+    """
+    return converter.structure_attrs_fromdict(_tr_data_keys(data), model_type)
+
+
+def forward_ref_structure_hook(context, converter, data, forward_ref):
+    """Applied to ForwardRef model and enum annotations
 
     - Map reserved words in json keys to approriate (safe) names in model.
     - handle ForwardRef types until github.com/Tinche/cattrs/pull/42/ is fixed
        Note: this is the reason we need a "context" param and have to use a
        partial func to register the hook. Once the issue is resolved we can
        remove "context" and the partial.
     """
-    for reserved in keyword.kwlist:
-        if reserved in data:
-            data[f"{reserved}_"] = data.pop(reserved)
-    if isinstance(type_, ForwardRef):
-        type_ = eval(type_.__forward_arg__, context, locals())
-    instance = converter.structure_attrs_fromdict(data, type_)
+    data = _tr_data_keys(data)
+    actual_type = eval(forward_ref.__forward_arg__, context, locals())
+    if issubclass(actual_type, enum.Enum):
+        instance = converter.structure(data, actual_type)
```

Comment:
handle `enum: "Enum"` type declarations in deserialization (well, with the help of the `__annotations__` hack converting that to `enum: ForwardRef("Enum")`)

Reply:
Sorry Python doesn't have a better built-in serialization solution!

Reply:
oh, me too. Python was my gold standard scripting language for a while but the Typing implementation and this lack of easy/builtin class-instance/json serialization has disillusioned me. I'd say Typescript is quickly gaining ground for me except the huge barrier between "I wrote this typescript code" and "I want to run this typescript script" (even as a node script) is still daunting to me.

Reply:
Yeah, the scripting in this repo is a bit awkward now, but if you look at the existing node scripts in the root package.json you can see the pattern that should always work.